### PR TITLE
ARQ-1937 added log warning and modified javadoc

### PR DESCRIPTION
**Pull request by MatousJobanek**
Hi,
I've modified the javadoc for the Deployer.deploy according to your comment here: https://github.com/MatousJobanek/arquillian-core/commit/42a2528cb65f540c5b6114deec338a1796f58901
I hope that it is OK now. 
If you think that changing the javadoc/adding the log warning is a nonsense in this case, just let me know ;-).

### CODE REVIEW

File: container/test-api/src/main/java/org/jboss/arquillian/container/test/api/Deployer.java [hunk 1]
```diff
@@ -51,6 +51,32 @@
    /**
     * Deploy the named deployment. <br/>
     * The operation will block until deploy is complete.
+    * <p>
```
Comment (smiklosovic) [inline container/test-api/src/main/java/org/jboss/arquillian/container/test/api/Deployer.java]: funny, I have never thought about this corner case ...

**Comment by aslakknutsen** [comment]
Thank you! Pushed upstream https://github.com/arquillian/arquillian-core/commit/96ad96f345e06c11e04eef94d0a3b4f5f23f684c
